.github/workflows/health.yaml

```diff
@@ -55,12 +55,30 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+        with:
+          ref: ${{ github.event.pull_request.base.ref }}
+          path: .coverage_base
+          
       - uses: dart-lang/setup-dart@d6a63dab3335f427404425de0fbfed4686d93c4f
         with:
           sdk: ${{ inputs.sdk }}
 
+      - name: Install coverage
+        run: dart pub global activate coverage
+
+      - name: Get old coverage
+        run: find . -name "pubspec.yaml" -exec dirname {} \; | while read -r line; do pushd .;echo $line;cd $line; dart pub get; [ -d test ] && dart pub global run coverage:test_with_coverage; popd; done 
```

Comment:
This is a bit complex, but I am no bash expert, so there is probably a much easier way to run `dart pub global run coverage:test_with_coverage` for all packages...